features/steps/steps.py

```diff
@@ -211,14 +239,59 @@ def instance_getter(i,representation_id, representation_type, negative=False):
         if condition(i, representation_id, representation_type):
             return i
 
+def strip_split(stmt, strp = ' ', splt = ' '):
+    return list(
+        map(str.lower, map(lambda s: s.strip(strp), stmt.split(splt)))
+    )
+
+def include_subtypes(stmt):
+    stmt = strip_split(stmt, strp = '[]')
+    if len(stmt) > 1 and 'subtypes' in stmt:
+        excluding_statements = ['without', 'not', 'excluding', 'no']
+        if len(set(stmt).intersection(set(excluding_statements))):
```

Comment:
1. len() is not necessary. empty set evaluates to false, same as 0 evaluates to false.
2. you can simply write: `return not set(stmt).intersection(set(excluding_statements))`. `if something: return False else return True` is considered a bit unecessary because you can in that case just return `not something`.
3. better write `excluding_statements = {'without', 'not', 'excluding', 'no'}` as a set literal so that you don't need to convert to set later on.

Reply:
That looks a lot cleaner. Implemented in https://github.com/buildingSMART/ifc-gherkin-rules/pull/35/commits/f002ab5cd3d2b969a63fcf05d16796dab2cee9fd